airflow/www/static/js/dag/details/dagRun/MarkRunAs.tsx

```diff
@@ -0,0 +1,91 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import {
+  Flex,
+  Button,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  MenuButtonProps,
+} from "@chakra-ui/react";
+import { MdArrowDropDown } from "react-icons/md";
+import { getMetaValue } from "src/utils";
+import { useMarkFailedRun, useMarkSuccessRun } from "src/api";
+import type { RunState } from "src/types";
+
+import { SimpleStatus } from "../../StatusBox";
+
+const canEdit = getMetaValue("can_edit") === "True";
+const dagId = getMetaValue("dag_id");
+
+interface Props extends MenuButtonProps {
+  runId: string;
+  state?: RunState;
+}
+
+const MarkRunAs = ({ runId, state, ...otherProps }: Props) => {
```

Comment:
Not sure why we need the extra `otherProps` propagated to the MenuButton child ? Doesn't look like we use it for now, I think it can be removed.